### PR TITLE
Add EVP_PKEY_get_hmac() function

### PR DESCRIPTION
**Pull request by npmccallum**
Before the addition of this function, it was impossible to read the
symmetric key from an EVP_PKEY_HMAC type EVP_PKEY.

### CODE REVIEW

**Comment by ekasper** [comment]
Why do you need to read this key?

File: crypto/evp/p_lib.c [hunk 1]
```diff
@@ -237,6 +237,17 @@ void *EVP_PKEY_get0(const EVP_PKEY *pkey)
     return pkey->pkey.ptr;
 }
 
+const unsigned char *EVP_PKEY_get_hmac(const EVP_PKEY *pkey, int *len)
```
Comment (ekasper) [inline crypto/evp/p_lib.c]: Should be `EVP_PKEY_get0_hmac` or perhaps better yet `EVP_PKEY_get0_hmac_key`?

size_t for the length.

The type check should happen before you call EVP_PKEY_get0.

Comment (richsalz) [inline crypto/evp/p_lib.c]: Does this stand a *chance* of making it into 1.1?  If not, we should be looking at other things ☺

Comment (ekasper) [inline crypto/evp/p_lib.c]: I think this falls in the "missing accessors now that stuff is opaque" category, so if it's really needed, yes it should make it to 1.1.0.

**Comment by npmccallum** [comment]
@ekasper https://github.com/npmccallum/jose

I'm writing a library for JOSE that wraps around OpenSSL. It provides functions to convert from JSON Web Keys (JWKs) to EVP_PKEYs and vice versa: https://github.com/npmccallum/jose/blob/master/src/jwk.h

I can convert from a JWK to a EVP_PKEY_HMAC, but not back since I have no direct access to the buffer: https://github.com/npmccallum/jose/blob/master/src/jwk.c#L156

I also need access to the length of the key in order to make a suggestion about what signing protocol to use: https://github.com/npmccallum/jose/blob/master/src/hmac.c#L10

File: crypto/evp/p_lib.c (hunk 1 above)
Comment (npmccallum) [inline crypto/evp/p_lib.c]: This is fixed in my latest commit.

Comment (kroeckx) [inline crypto/evp/p_lib.c]: It seems to be more consistent to use EVP_PKEY_get0_hmac(), without _key.

Comment (npmccallum) [inline crypto/evp/p_lib.c]: I don't care which function name we use. Just let me know what the consensus is and I'll change it.

Comment (ekasper) [inline crypto/evp/p_lib.c]: Go with `EVP_PKEY_get0_hmac` then, fine by me.

**Comment by ekasper** [comment]
Thanks for the clarification, that's reasonable. +1 from me.

File: crypto/evp/p_lib.c (hunk 1 above)
Comment (npmccallum) [inline crypto/evp/p_lib.c]: The latest push uses EVP_PKEY_get0_hmac().

**Comment by npmccallum** [comment]
Can this be merged now that I have addressed all of the stated concerns?

**Comment by richsalz** [comment]
+1 i will get to it shortly.

**Comment by npmccallum** [comment]
AFAICS, this is closed.

**Pull request closed without merging**